### PR TITLE
Calcite master

### PR DESCRIPTION
**Pull request by yzhang1991**

### CODE REVIEW

File: src/frontend/org/voltdb/sysprocs/AdHoc.java [hunk 1]
```diff
@@ -26,15 +26,93 @@
...
+                // No mixing DDL and DML/DQL. Turn this into an error and return it to the client.
+                return makeQuickResponse(
+                        ClientResponse.GRACEFUL_FAILURE,
+                        "DDL mixed with DML and queries is unsupported.");
+            }
+            rootNodesOfParsedQueries.add(sqlNode);
+        }
+
+        if (isDDLBatch) {
+            return runDDLBatch(sqlList);
```
Comment (kaikaizi) [inline src/frontend/org/voltdb/sysprocs/AdHoc.java]: It's better if you pass both `sqlList` and `rootNodesOfParsedQueries` to both functions. For now, we ignore `rootNodesOfParsedQueries`.

File: template.classpath [hunk 2]
```diff
@@ -74,9 +74,13 @@
 	<classpathentry kind="lib" path="lib/scala-parser-combinators_2.11-1.0.2.jar"/>
 	<classpathentry kind="lib" path="lib/scala-xml_2.11-1.0.2.jar"/>
 	<classpathentry kind="lib" path="third_party/java/jars/javassist-3.20.0-GA.jar"/>
-    <!-- zkclient is used for test only -->
+	<classpathentry kind="lib" path="lib/calcite-core-1.17.0.jar" sourcepath="lib/calcite-core-1.17.0-sources.jar"/>
+	<classpathentry kind="lib" path="lib/calcite-server-1.17.0.jar" sourcepath="lib/calcite-server-1.17.0-sources.jar"/>
+	<classpathentry kind="lib" path="lib/calcite-linq4j-1.17.0.jar" sourcepath="lib/calcite-linq4j-1.17.0-sources.jar"/>
+	<classpathentry kind="lib" path="lib/avatica-core-1.12.0.jar" sourcepath="lib/avatica-core-1.12.0-sources.jar"/>
```
Comment (Chaozz) [inline template.classpath]: Do we need this avatica lib now or in the future?

File: src/frontend/org/voltdb/sysprocs/AdHoc.java (hunk 1 above)
Comment (yzhang1991) [reply src/frontend/org/voltdb/sysprocs/AdHoc.java]: I created `runDDLBatchThroughCalcite()` and `runNonDDLAdHocThroughCalcite()` for us to replace when they are ready.

**Comment by yzhang1991** [comment]
This branch is the feature branch for the Calcite project. This pull request is for review only, the changes will not go directly to master for now.

**Pull request closed without merging**